src/test/kotlin/edu/carroll/polyominoes/service/account/RegisterServiceImplTest.kt

```diff
@@ -0,0 +1,289 @@
+package edu.carroll.polyominoes.service.account
+
+import jakarta.transaction.Transactional
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+
+@Transactional
+@SpringBootTest
+internal class RegisterServiceImplTest(
+        @Autowired private val registerServiceImpl: RegisterServiceImpl
+) {
+
+    companion object {
+
+        private val testAccount0 = TestingAccount("testing.user", "test@carroll.edu", "testing123!") //existing
+        private val testAccount1 = TestingAccount("testing.user1", "test1@carroll.edu", "testing456!")
+        private val testAccount2 = TestingAccount("testing.user2", "test2@gmail.com", "testing7890!")
+    }
+
+    internal class TestingAccount {
+        var username: String
+        var email: String
+        var rawPassword: String
+
+        constructor(username : String, email : String, rawPassword : String) {
+            this.username = username
+            this.email = email
+            this.rawPassword = rawPassword
+        }
+    }
+
+
+    @Test
+    fun createUserOneUserSuccessTest() {
+        assertTrue(registerServiceImpl.createUser(testAccount1.username, testAccount1.email, testAccount1.rawPassword))
+        assertTrue(registerServiceImpl.validateUsernameExist(testAccount1.username))
+        assertTrue(registerServiceImpl.validateEmailExist(testAccount1.email))
+    }
+
+
+    @Test
+    fun createUserTwoUserSuccessTest() {
+        assertTrue(registerServiceImpl.createUser(testAccount1.username, testAccount1.email, testAccount1.rawPassword))
+        assertTrue(registerServiceImpl.createUser(testAccount2.username, testAccount2.email, testAccount2.rawPassword))
+
+        // testAccount1
+        assertTrue(registerServiceImpl.validateUsernameExist(testAccount1.username))
+        assertTrue(registerServiceImpl.validateEmailExist(testAccount1.email))
+
+        // testAccount2
+        assertTrue(registerServiceImpl.validateUsernameExist(testAccount2.username))
+        assertTrue(registerServiceImpl.validateEmailExist(testAccount2.email))
+    }
+
+    @Test
+    fun createUserUsernameExistsTest() {
+        assertTrue(registerServiceImpl.createUser(testAccount0.username, testAccount0.email, testAccount0.rawPassword))
+        assertFalse(registerServiceImpl.createUser(testAccount0.username, testAccount1.email, testAccount1.rawPassword))
+    }
```

Comment:
A true/false response is a good start, but you should also ensure that the accounts were also created/not-created using a separate call (validUsernameExist/validateEmailExists), because the method could return the 'right' return type, but not actually do the right thing.